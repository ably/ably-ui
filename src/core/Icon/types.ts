import { computedIcons } from "./computed-icons";

export const iconNames = {
  gui: computedIcons.gui,
  display: computedIcons.display,
  social: computedIcons.social,
  other: computedIcons.other,
  tech: computedIcons.tech,
  product: computedIcons.product,
};

export type IconName =
  | (typeof iconNames.gui)[number]
  | (typeof iconNames.display)[number]
  | (typeof iconNames.social)[number]
  | (typeof iconNames.other)[number]
  | (typeof iconNames.tech)[number]
  | (typeof iconNames.product)[number];
