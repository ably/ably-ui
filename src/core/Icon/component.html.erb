<%= tag.svg class: "#{color} #{additional_css}", style: "width: #{size}; height: #{size}", **additional_attributes do %>
  <%= tag.use "xlink:href": "#sprite-#{name}" %>
<% end %>
