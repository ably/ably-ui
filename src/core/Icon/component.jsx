import React from "react";
import T from "prop-types";

const Icon = ({ name, size = "0.75rem", color = "", additionalCSS = "", ...additionalAttributes }) => {
  return (
    <svg className={`${color} ${additionalCSS}`} style={{ width: size, height: size }} {...additionalAttributes}>
      <use xlinkHref={`#sprite-${name}`} />
    </svg>
  );
};

Icon.propTypes = {
  name: T.string.isRequired,
  size: T.string,
  color: T.string,
  additionalCSS: T.string,
};

export default Icon;
