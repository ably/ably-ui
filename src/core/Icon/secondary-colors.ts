import { ColorClass } from "../styles/colors/types";
import { IconName } from "./types";

export const defaultIconSecondaryColor = (name: IconName) => {
  const colorLookup = {
    "icon-gui-check-circled-fill": "text-white",
  } as Record<IconName, ColorClass>;

  return colorLookup[name];
};
