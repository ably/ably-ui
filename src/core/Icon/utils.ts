import { ColorClass } from "../styles/colors/types";
import { IconName } from "./types";

export const defaultIconSecondaryColor = (name: IconName) => {
  const colorLookup = {
    "icon-gui-check-circled-fill": "text-white",
  } as Record<string, ColorClass>;

  return colorLookup[name];
};

// This is a mapping of older icon names to their newer HeroIcon counterparts.
export const guiIconAliases: Record<string, string> = {
  "icon-gui-arrow-bidirectional-horizontal":
    "icon-gui-arrows-right-left-outline",
  "icon-gui-arrow-bidirectional-vertical": "icon-gui-arrows-up-down-outline",
  "icon-gui-arrow-down": "icon-gui-arrow-long-down-outline",
  "icon-gui-arrow-left": "icon-gui-arrow-long-left-outline",
  "icon-gui-arrow-right": "icon-gui-arrow-long-right-outline",
  "icon-gui-arrow-up": "icon-gui-arrow-long-up-outline",
  "icon-gui-burger-menu": "icon-gui-bars-3-outline",
  "icon-gui-calendar": "icon-gui-calendar-days-outline",
  "icon-gui-check": "icon-gui-check-outline",
  "icon-gui-check-circled": "icon-gui-check-circle-outline",
  "icon-gui-clock": "icon-gui-clock-outline",
  "icon-gui-close": "icon-gui-x-mark-outline",
  "icon-gui-code-symbol": "icon-gui-code-bracket-outline",
  "icon-gui-copy": "icon-gui-square-2-stack-outline",
  "icon-gui-cross-circled-fill": "icon-gui-x-circle-solid",
  "icon-gui-cross-circled": "icon-gui-x-circle-outline",
  "icon-gui-dash-circled": "icon-gui-minus-circle-outline",
  "icon-gui-disclosure-arrow-down": "icon-gui-chevron-down-outline",
  "icon-gui-disclosure-arrow-left": "icon-gui-chevron-left-outline",
  "icon-gui-disclosure-arrow-right": "icon-gui-chevron-right-outline",
  "icon-gui-disclosure-arrow-up": "icon-gui-chevron-up-outline",
  "icon-gui-disclosure-arrow": "icon-gui-chevron-right-outline",
  "icon-gui-document-generic": "icon-gui-document-text-outline",
  "icon-gui-download": "icon-gui-arrow-down-tray-outline",
  "icon-gui-enlarge": "icon-gui-magnifying-glass-plus-outline",
  "icon-gui-external-link": "icon-gui-arrow-top-right-on-square-outline",
  "icon-gui-folder": "icon-gui-folder-outline",
  "icon-gui-globe": "icon-gui-globe-alt-outline",
  "icon-gui-heart": "icon-gui-heart-outline",
  "icon-gui-home": "icon-gui-home-outline",
  "icon-gui-info": "icon-gui-information-circle-outline",
  "icon-gui-link-arrow": "icon-gui-arrow-long-right-outline",
  "icon-gui-map-pin": "icon-gui-map-pin-outline",
  "icon-gui-minus": "icon-gui-minus-outline",
  "icon-gui-padlock": "icon-gui-lock-closed-outline",
  "icon-gui-plus": "icon-gui-plus-outline",
  "icon-gui-quote-marks-solid": "icon-gui-quote-marks-fill",
  "icon-gui-search": "icon-gui-magnifying-glass-outline",
  "icon-gui-sliders": "icon-gui-adjustments-horizontal-outline",
  "icon-gui-stack": "icon-gui-square-3-stack-3d-outline",
  "icon-gui-star": "icon-gui-star-outline",
  "icon-gui-system-status-circled-fill": "icon-gui-arrow-up-circle-solid",
  "icon-gui-tick": "icon-gui-check-outline",
  "icon-gui-warning-fill": "icon-gui-exclamation-triangle-solid",
  "icon-gui-warning": "icon-gui-exclamation-triangle-outline",
};
