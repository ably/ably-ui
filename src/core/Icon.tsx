import React, { CSSProperties, useCallback, useId, useMemo } from "react";

import {
  defaultIconSecondaryColor,
  getHeroicon,
  setUniqueIds,
  toPascalCase,
} from "./Icon/utils";
import * as IconComponents from "./Icon/components";
import { IconName, IconSize } from "./Icon/types";
import { ColorClass, ColorThemeSet } from "./styles/colors/types";
import { convertTailwindClassToVar } from "./styles/colors/utils";
import cn from "./utils/cn";

export type IconProps = {
  name: IconName;
  size?: IconSize;
  color?: ColorClass | ColorThemeSet;
  secondaryColor?: ColorClass | ColorThemeSet;
  additionalCSS?: string;
  variant?: "outline" | "solid" | "micro" | "mini";
};

// Helper function to get a heroicon component from the static map
const getHeroiconComponent = (
  iconName: string,
): React.ComponentType<React.SVGProps<SVGSVGElement>> | null => {
  // Parse the icon name to extract heroicon name and variant
  // Expected format: icon-gui-{heroicon-name}-{variant}
  const match = iconName.match(/^icon-gui-(.+)-(outline|solid|mini|micro)$/);
  if (!match) {
    return null;
  }

  const [, heroiconName, variant] = match;
  const component = getHeroicon(heroiconName, variant);
  return component;
};

const Icon = ({
  name,
  size = "0.75rem",
  color,
  secondaryColor,
  additionalCSS = "",
  variant,
  ...additionalAttributes
}: IconProps) => {
  const uniqueId = useId();

  const [lightSecondaryColor, darkSecondaryColor] = (
    secondaryColor ?? ""
  ).split(" dark:") as [ColorClass, ColorClass | undefined];

  const iconName =
    variant && !name.endsWith(`-${variant}`) ? `${name}-${variant}` : name;

  // Get the local icon component (memoized)
  const LocalIconComponent = useMemo(():
    | React.ComponentType<React.SVGProps<SVGSVGElement>>
    | undefined => {
    return IconComponents[
      toPascalCase(iconName) as keyof typeof IconComponents
    ];
  }, [iconName]);

  // Get the heroicon component only if no local component is found (memoized)
  const HeroiconComponent = useMemo(() => {
    return LocalIconComponent ? null : getHeroiconComponent(iconName);
  }, [LocalIconComponent, iconName]);

  const setUniqueIdsRef = useCallback(
    (el: SVGSVGElement | null) => {
      setUniqueIds(el, uniqueId);
    },
    [uniqueId],
  );

  const iconSvg = useCallback(
    (secondaryColor?: ColorClass, isDark?: boolean, isThemed?: boolean) => {
      if (LocalIconComponent) {
        let secondaryColorValue;
        if (secondaryColor) {
          secondaryColorValue = convertTailwindClassToVar(secondaryColor);
        } else if (defaultIconSecondaryColor(name)) {
          secondaryColorValue = convertTailwindClassToVar(
            defaultIconSecondaryColor(name),
          );
        }

        return (
          <LocalIconComponent
            className={cn(
              { [`${color}`]: color },
              { [`${additionalCSS}`]: additionalCSS },
              {
                "hidden dark:block": secondaryColor && !isDark && isThemed,
              },
              {
                "dark:hidden": secondaryColor && isDark && isThemed,
              },
            )}
            style={
              {
                width: size,
                height: size,
                ...(secondaryColorValue && {
                  "--ui-icon-secondary-color": secondaryColorValue,
                }),
              } as CSSProperties
            }
            {...additionalAttributes}
            ref={setUniqueIdsRef}
          />
        );
      }

      if (HeroiconComponent) {
        return (
          <HeroiconComponent
            className={cn(
              { [`${color}`]: color },
              { [`${additionalCSS}`]: additionalCSS },
            )}
            style={{
              width: size,
              height: size,
            }}
            {...additionalAttributes}
          />
        );
      }
    },
    [
      LocalIconComponent,
      HeroiconComponent,
      iconName,
      size,
      color,
      additionalCSS,
      lightSecondaryColor,
      darkSecondaryColor,
      name,
    ],
  );

  if (lightSecondaryColor && darkSecondaryColor) {
    return (
      <>
        {iconSvg(lightSecondaryColor, false, !!darkSecondaryColor)}
        {iconSvg(darkSecondaryColor, true, true)}
      </>
    );
  } else if (lightSecondaryColor) {
    return iconSvg(lightSecondaryColor, false, !!darkSecondaryColor);
  } else {
    return iconSvg();
  }
};

export default Icon;
