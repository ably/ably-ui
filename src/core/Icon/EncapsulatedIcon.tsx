import React from "react";
import Icon, { IconProps } from "../Icon";
import useTheming from "../hooks/useTheming";
import { Theme } from "../styles/colors/types";
import { IconSize } from "./types";

type EncapsulatedIconProps = {
  theme?: Theme;
  className?: string;
  innerClassName?: string;
  iconSize?: IconSize;
} & IconProps;

const ICON_SIZE_REDUCTION = 12;
const ICON_HEIGHT_REDUCTION = 2;

const EncapsulatedIcon = ({
  theme = "dark",
  size = "40px",
  iconSize,
  className,
  innerClassName,
  ...iconProps
}: EncapsulatedIconProps) => {
  const { themeColor } = useTheming({
    baseTheme: "dark",
    theme,
  });
  let computedIconSize = size,
    computedIconHeight = size;

  if (iconSize) {
    computedIconSize = iconSize;
  } else if (size.toString().endsWith("px")) {
    const numericalSize = parseInt(size, 10);
    computedIconSize = `${numericalSize - ICON_SIZE_REDUCTION}px`;
    computedIconHeight = `${numericalSize - ICON_HEIGHT_REDUCTION}px`;
  } else {
    console.warn(
      `EncapsulatedIcon: Non-pixel units might not behave as expected`,
    );
  }

  return (
    <div
      className={`p-1 rounded-lg ${theme === "light" ? "bg-gradient-to-t" : "bg-gradient-to-b"} ${themeColor("from-neutral-900")} ${className ?? ""}`}
      style={{ width: size, height: size }}
    >
      <div
        className={`flex items-center justify-center rounded-lg ${themeColor("bg-neutral-1100")} ${innerClassName ?? ""}`}
        style={{ height: computedIconHeight }}
      >
        <Icon size={computedIconSize} {...iconProps} />
      </div>
    </div>
  );
};

export default EncapsulatedIcon;
