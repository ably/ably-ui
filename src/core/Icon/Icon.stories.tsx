import React from "react";
import Icon from "../Icon";

export default {
  title: "Components/Icon",
  component: Icon,
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
};

const coreIcons = [
  "icon-gui-ably-badge",
  "icon-gui-arrow-bidirectional-horizontal",
  "icon-gui-arrow-bidirectional-vertical",
  "icon-gui-arrow-down",
  "icon-gui-arrow-left",
  "icon-gui-arrow-right",
  "icon-gui-arrow-up",
  "icon-gui-burger-menu",
  "icon-gui-check-circled-fill",
  "icon-gui-check-circled-fill-black",
  "icon-gui-check-circled",
  "icon-gui-checklist-checked",
  "icon-gui-clock",
  "icon-gui-close",
  "icon-gui-cross-circled-fill",
  "icon-gui-cross-circled",
  "icon-gui-copy",
  "icon-gui-dash-circled",
  "icon-gui-disclosure-arrow",
  "icon-gui-document-generic",
  "icon-gui-enlarge",
  "icon-gui-external-link",
  "icon-gui-history",
  "icon-gui-info",
  "icon-gui-link-arrow",
  "icon-gui-live-chat",
  "icon-gui-minus",
  "icon-gui-plus",
  "icon-gui-quote-marks-solid",
  "icon-gui-refresh",
  "icon-gui-search",
  "icon-gui-tick",
  "icon-gui-warning",
  "icon-gui-link",
  "icon-gui-filter-flow-step-1",
  "icon-gui-filter-flow-step-2",
  "icon-gui-filter-flow-step-3",
  "icon-gui-resources",
];

const displayIcons = [
  "icon-display-48hrs",
  "icon-display-about-ably-col",
  "icon-display-api",
  "icon-display-api-keys",
  "icon-display-asset-tracking-col",
  "icon-display-browser",
  "icon-display-calendar",
  "icon-display-call-mobile",
  "icon-display-careers-col",
  "icon-display-case-studies-col",
  "icon-display-chat-stack-col",
  "icon-display-cloud-servers",
  "icon-display-compare-tech-col",
  "icon-display-customers-col",
  "icon-display-docs-col",
  "icon-display-documentation",
  "icon-display-examples-col",
  "icon-display-gdpr",
  "icon-display-general-comms",
  "icon-display-hipaa",
  "icon-display-it-support-access",
  "icon-display-it-support-helpdesk",
  "icon-display-laptop",
  "icon-display-lightbulb-col",
  "icon-display-live-chat",
  "icon-display-map-pin",
  "icon-display-message",
  "icon-display-padlock-closed",
  "icon-display-platform",
  "icon-display-play",
  "icon-display-privacy-shield-framework",
  "icon-display-quickstart-guides-col",
  "icon-display-resources-col",
  "icon-display-sdks-col",
  "icon-display-servers",
  "icon-display-shopping-cart",
  "icon-display-sla",
  "icon-display-chat-col",
  "icon-display-soc2-type2",
  "icon-display-tech-account-comms",
  "icon-display-tutorials-demos-col",
  "icon-display-integrations-col",
  "icon-display-virtual-events-col",
  "icon-live-updates-results-metrics-col",
  "icon-multi-user-spaces-col",
];

const socialIcons = [
  "discord",
  "facebook",
  "github",
  "glassdoor",
  "linkedin",
  "twitter",
  "google",
  "stackoverflow",
  "youtube",
];

const otherIcons = ["quote"];

const renderIcons = (iconSet: string[]) => (
  <div className="grid grid-cols-4 ui-grid-gap max-w-screen-lg mb-64">
    {iconSet.map((icon: string) => (
      <div className="border p-16" key={icon}>
        <code className="ui-text-code mb-8 block">{icon}</code>
        <div className="border inline-flex flex-0">
          <div className="flex pi-checkered-bg">
            <Icon
              name={icon}
              additionalCSS="hover:text-active-orange"
              color="text-cool-black"
              size="1.5rem"
            />
          </div>
        </div>
      </div>
    ))}
  </div>
);

export const GUIIcons = {
  render: () => renderIcons(coreIcons),
};

export const DisplayIcons = {
  render: () => renderIcons(displayIcons),
};

export const SocialIcons = {
  render: () => renderIcons(socialIcons),
};

export const Other = {
  render: () => renderIcons(otherIcons),
};
