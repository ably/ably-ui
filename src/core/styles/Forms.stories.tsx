import React from "react";
import Icon from "../Icon";

export default {
  title: "CSS/Forms",
};

const exampleOptionFields = (
  <>
    <option value="select" disabled>
      Select language
    </option>
    <option value="javascript">Javascript</option>
    <option value="ruby">Ruby</option>
    <option value="python">Python</option>
    <option value="java">Java</option>
  </>
);

export const Inputs = {
  render: () => (
    <div className="grid sm:grid-cols-2 gap-16">
      <div>
        <p className="mb-16 text-neutral-800">Default</p>
        <input
          type="search"
          className="ui-input"
          placeholder="Light mode input"
          autoComplete="off"
        />
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Default (dark)</p>
        <input
          type="search"
          className="ui-input ui-input-dark"
          placeholder="Dark mode input"
          autoComplete="off"
        />
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Disabled</p>
        <input
          type="search"
          className="ui-input disabled"
          placeholder="Light mode input"
          autoComplete="off"
          disabled
        />
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Disabled (dark)</p>
        <input
          type="search"
          className="ui-input ui-input-dark"
          placeholder="Dark mode input"
          autoComplete="off"
          disabled
        />
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Invalid</p>
        <input
          type="search"
          className="ui-input disabled"
          placeholder="Light mode input"
          autoComplete="off"
          required
        />
        <p className="ui-text-p3 text-gui-error-red mt-8">Oof, what an input</p>
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Invalid (dark)</p>
        <input
          type="search"
          className="ui-input ui-input-dark"
          placeholder="Dark mode input"
          autoComplete="off"
          required
        />
        <p className="ui-text-p3 text-gui-error-red mt-8">Oof, what an input</p>
      </div>
      <div>
        <p className="mb-16 text-neutral-800">With character insert</p>
        <div className="relative">
          <div className="h-32 w-32 absolute left-8 top-8 flex items-center justify-center select-none cursor-default">
            $
          </div>
          <input
            type="search"
            className="ui-input pl-40"
            placeholder="With icon"
            autoComplete="off"
          />
        </div>
      </div>
    </div>
  ),
};

export const Dropdowns = {
  render: () => (
    <div className="grid sm:grid-cols-2 gap-16">
      <div>
        <p className="mb-16 text-neutral-800">Default</p>
        <select className="ui-dropdown" defaultValue="select">
          {exampleOptionFields}
        </select>
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Small</p>
        <select className="ui-dropdown-small" defaultValue="select">
          {exampleOptionFields}
        </select>
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Disabled</p>
        <select className="ui-dropdown" defaultValue="select" disabled>
          {exampleOptionFields}
        </select>
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Synthetic</p>
        <div className="ui-dropdown max-w-256" tabIndex={0}>
          Select language
        </div>
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Dark mode</p>
        <div className="bg-neutral-1300 p-24">
          <select
            className="ui-dropdown ui-dropdown-dark"
            defaultValue="select"
          >
            {exampleOptionFields}
          </select>
        </div>
      </div>
    </div>
  ),
};

export const Checkboxes = {
  render: () => (
    <div className="grid sm:grid-cols-2 gap-16">
      <div>
        <p className="mb-16 text-neutral-800">Default</p>
        <div className="ui-checkbox-p1">
          <input
            data-ui-checkbox-native=""
            type="checkbox"
            id="checkbox-example-1"
            name="checkbox-example-1"
            value="yes"
            className="ui-checkbox-input"
          />
          <div data-ui-checkbox-styled="" className="ui-checkbox-styled">
            <Icon
              name="icon-gui-tick"
              size="1rem"
              additionalCSS="ui-checkbox-styled-tick"
            />
          </div>
          <label htmlFor="checkbox-example-1" className="ui-checkbox-label-p1">
            Subscribe to the Ably newsletter
          </label>
        </div>
      </div>
      <div>
        <p className="mb-16 text-neutral-800">Disabled</p>
        <div className="ui-checkbox-p1">
          <input
            data-ui-checkbox-native=""
            type="checkbox"
            id="checkbox-example-2"
            name="checkbox-example-2"
            value="yes"
            className="ui-checkbox-input"
            disabled
          />
          <div data-ui-checkbox-styled="" className="ui-checkbox-styled">
            <Icon
              name="icon-gui-tick"
              size="1rem"
              additionalCSS="ui-checkbox-styled-tick"
            />
          </div>
          <label htmlFor="checkbox-example-2" className="ui-checkbox-label-p2">
            Subscribe to the Ably newsletter
          </label>
        </div>
      </div>
    </div>
  ),
};

export const Textareas = {
  render: () => (
    <div className="mb-40">
      <label className="ui-text-p1 block" htmlFor="example-text-area"></label>
      <textarea
        id="example-text-area"
        className="ui-textarea"
        rows={2}
        placeholder="The <textarea> HTML element represents a multi-line plain-text editing control, useful when you want to allow users to enter a sizeable amount of free-form text, for example a comment on a review or feedback form."
      ></textarea>
      <p className="ui-text-p3 text-dark-grey">
        Example of additional text beneath the textarea.
      </p>
    </div>
  ),
};

export const Toggles = {
  render: () => (
    <div className="grid sm:grid-cols-2 gap-16">
      <div className="flex gap-16 items-center">
        <label className="ui-toggle">
          <input type="checkbox" />
          <span className="ui-toggle-slider" />
        </label>
        <p>Default</p>
      </div>
      <div className="flex gap-16 items-center">
        <label className="ui-toggle">
          <input type="checkbox" defaultChecked />
          <span className="ui-toggle-slider" />
        </label>
        <p>Pre-checked</p>
      </div>
      <div className="flex gap-16 items-center">
        <label className="ui-toggle">
          <input type="checkbox" disabled />
          <span className="ui-toggle-slider" />
        </label>
        <p>Disabled</p>
      </div>
      <div className="flex gap-16 items-center">
        <label className="ui-toggle">
          <input type="checkbox" className="peer" />
          <span className="ui-toggle-slider peer-checked:!bg-pink-600 bg-blue-800" />
        </label>
        <p>Custom color</p>
      </div>
    </div>
  ),
};
