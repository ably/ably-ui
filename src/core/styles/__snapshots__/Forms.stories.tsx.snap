// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles/Forms Checkboxes smoke-test 1`] = `
<div class="grid sm:grid-cols-2 gap-4">
  <div class="flex flex-col gap-2">
    <p class="mb-4 text-neutral-800">
      Default
    </p>
    <div class="ui-checkbox-p1 flex items-center">
      <input data-ui-checkbox-native
             type="checkbox"
             id="checkbox-example-1"
             name="checkbox-example-1"
             class="ui-checkbox-input"
             value="yes"
      >
      <div data-ui-checkbox-styled
           class="ui-checkbox-styled"
      >
        <svg class="inline-block ui-checkbox-styled-tick"
             style="width: 1rem; height: 1rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-micro">
          </use>
        </svg>
      </div>
      <label for="checkbox-example-1"
             class="ui-checkbox-label-p1"
      >
        Option 1
      </label>
    </div>
    <div class="ui-checkbox-p1 flex items-center">
      <input data-ui-checkbox-native
             type="checkbox"
             id="checkbox-example-2"
             name="checkbox-example-2"
             class="ui-checkbox-input"
             value="yes"
      >
      <div data-ui-checkbox-styled
           class="ui-checkbox-styled"
      >
        <svg class="inline-block ui-checkbox-styled-tick"
             style="width: 1rem; height: 1rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-micro">
          </use>
        </svg>
      </div>
      <label for="checkbox-example-2"
             class="ui-checkbox-label-p1"
      >
        Option 2
      </label>
    </div>
    <div class="ui-checkbox-p1 flex items-center">
      <input data-ui-checkbox-native
             type="checkbox"
             id="checkbox-example-3"
             name="checkbox-example-3"
             class="ui-checkbox-input"
             value="yes"
      >
      <div data-ui-checkbox-styled
           class="ui-checkbox-styled"
      >
        <svg class="inline-block ui-checkbox-styled-tick"
             style="width: 1rem; height: 1rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-micro">
          </use>
        </svg>
      </div>
      <label for="checkbox-example-3"
             class="ui-checkbox-label-p1"
      >
        Option 3
      </label>
    </div>
  </div>
  <div class="flex flex-col gap-2">
    <p class="mb-4 text-neutral-800">
      Disabled
    </p>
    <div class="ui-checkbox-p1 flex items-center">
      <input data-ui-checkbox-native
             type="checkbox"
             id="checkbox-example-4"
             name="checkbox-example-4"
             class="ui-checkbox-input"
             disabled
             value="yes"
      >
      <div data-ui-checkbox-styled
           class="ui-checkbox-styled"
      >
        <svg class="inline-block ui-checkbox-styled-tick"
             style="width: 1rem; height: 1rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-micro">
          </use>
        </svg>
      </div>
      <label for="checkbox-example-4"
             class="ui-checkbox-label-p1"
      >
        Option 4
      </label>
    </div>
    <div class="ui-checkbox-p1 flex items-center">
      <input data-ui-checkbox-native
             type="checkbox"
             id="checkbox-example-5"
             name="checkbox-example-5"
             class="ui-checkbox-input"
             disabled
             value="yes"
      >
      <div data-ui-checkbox-styled
           class="ui-checkbox-styled"
      >
        <svg class="inline-block ui-checkbox-styled-tick"
             style="width: 1rem; height: 1rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-micro">
          </use>
        </svg>
      </div>
      <label for="checkbox-example-5"
             class="ui-checkbox-label-p1"
      >
        Option 5
      </label>
    </div>
    <div class="ui-checkbox-p1 flex items-center">
      <input data-ui-checkbox-native
             type="checkbox"
             id="checkbox-example-6"
             name="checkbox-example-6"
             class="ui-checkbox-input"
             disabled
             value="yes"
      >
      <div data-ui-checkbox-styled
           class="ui-checkbox-styled"
      >
        <svg class="inline-block ui-checkbox-styled-tick"
             style="width: 1rem; height: 1rem;"
        >
          <use xlink:href="#sprite-icon-gui-check-micro">
          </use>
        </svg>
      </div>
      <label for="checkbox-example-6"
             class="ui-checkbox-label-p1"
      >
        Option 6
      </label>
    </div>
  </div>
</div>
`;

exports[`Styles/Forms Dropdowns smoke-test 1`] = `
<div class="grid sm:grid-cols-2 gap-4">
  <div>
    <p class="mb-4 text-neutral-800">
      Default
    </p>
    <select class="ui-dropdown">
      <option value="select"
              disabled
              selected
      >
        Select language
      </option>
      <option value="javascript">
        Javascript
      </option>
      <option value="ruby">
        Ruby
      </option>
      <option value="python">
        Python
      </option>
      <option value="java">
        Java
      </option>
    </select>
  </div>
  <div>
    <p class="mb-4 text-neutral-800">
      Small
    </p>
    <select class="ui-dropdown-small">
      <option value="select"
              disabled
              selected
      >
        Select language
      </option>
      <option value="javascript">
        Javascript
      </option>
      <option value="ruby">
        Ruby
      </option>
      <option value="python">
        Python
      </option>
      <option value="java">
        Java
      </option>
    </select>
  </div>
  <div>
    <p class="mb-4 text-neutral-800">
      Disabled
    </p>
    <select class="ui-dropdown"
            disabled
    >
      <option value="select"
              disabled
              selected
      >
        Select language
      </option>
      <option value="javascript">
        Javascript
      </option>
      <option value="ruby">
        Ruby
      </option>
      <option value="python">
        Python
      </option>
      <option value="java">
        Java
      </option>
    </select>
  </div>
  <div>
    <p class="mb-4 text-neutral-800">
      Synthetic
    </p>
    <div class="ui-dropdown max-w-64"
         tabindex="0"
    >
      Select language
    </div>
  </div>
</div>
`;

exports[`Styles/Forms Inputs smoke-test 1`] = `
<div class="grid sm:grid-cols-2 gap-4">
  <div>
    <p class="mb-4 text-neutral-800">
      Default
    </p>
    <input type="search"
           class="ui-input"
           placeholder="Light mode input"
           autocomplete="off"
    >
  </div>
  <div>
    <p class="mb-4 text-neutral-800">
      Disabled
    </p>
    <input type="search"
           class="ui-input disabled"
           placeholder="Light mode input"
           autocomplete="off"
           disabled
    >
  </div>
  <div>
    <p class="mb-4 text-neutral-800">
      Invalid
    </p>
    <input type="search"
           class="ui-input disabled"
           placeholder="Light mode input"
           autocomplete="off"
           required
    >
    <p class="ui-text-p3 text-gui-error-red mt-2">
      Oof, what an input
    </p>
  </div>
  <div>
    <p class="mb-4 text-neutral-800">
      With character insert
    </p>
    <div class="relative">
      <div class="h-8 w-8 absolute left-2 top-2 flex items-center justify-center select-none cursor-default">
        $
      </div>
      <input type="search"
             class="ui-input pl-10"
             placeholder="With icon"
             autocomplete="off"
      >
    </div>
  </div>
</div>
`;

exports[`Styles/Forms RadioButtons smoke-test 1`] = `
<div class="grid sm:grid-cols-2 gap-4">
  <div class="flex flex-col gap-2">
    <p class="mb-4 text-neutral-800">
      Default
    </p>
    <div class="flex items-center gap-2">
      <input data-ui-radio-native
             type="radio"
             id="radio-example-1"
             name="radio-example"
             class="ui-radio"
             value="option-1"
      >
      <label class="text-neutral-1300 dark:text-neutral-000"
             for="radio-example-1"
      >
        Option 1
      </label>
    </div>
    <div class="flex items-center gap-2">
      <input data-ui-radio-native
             type="radio"
             id="radio-example-2"
             name="radio-example"
             class="ui-radio"
             value="option-2"
      >
      <label class="text-neutral-1300 dark:text-neutral-000"
             for="radio-example-2"
      >
        Option 2
      </label>
    </div>
    <div class="flex items-center gap-2">
      <input data-ui-radio-native
             type="radio"
             id="radio-example-3"
             name="radio-example"
             class="ui-radio"
             value="option-3"
      >
      <label class="text-neutral-1300 dark:text-neutral-000"
             for="radio-example-3"
      >
        Option 3
      </label>
    </div>
  </div>
  <div class="flex flex-col gap-2">
    <p class="mb-4 text-neutral-800">
      Disabled
    </p>
    <div class="flex items-center gap-2">
      <input data-ui-radio-native
             type="radio"
             id="radio-example-4"
             name="radio-example"
             class="ui-radio"
             disabled
             value="option-4"
      >
      <label class="text-neutral-1300 dark:text-neutral-000"
             for="radio-example-4"
      >
        Option 4
      </label>
    </div>
    <div class="flex items-center gap-2">
      <input data-ui-radio-native
             type="radio"
             id="radio-example-5"
             name="radio-example"
             class="ui-radio"
             disabled
             value="option-5"
      >
      <label class="text-neutral-1300 dark:text-neutral-000"
             for="radio-example-5"
      >
        Option 5
      </label>
    </div>
    <div class="flex items-center gap-2">
      <input data-ui-radio-native
             type="radio"
             id="radio-example-6"
             name="radio-example"
             class="ui-radio"
             disabled
             value="option-6"
      >
      <label class="text-neutral-1300 dark:text-neutral-000"
             for="radio-example-6"
      >
        Option 6
      </label>
    </div>
  </div>
</div>
`;

exports[`Styles/Forms Textareas smoke-test 1`] = `
<div class="mb-10">
  <label class="ui-text-p1 block"
         for="example-text-area"
  >
  </label>
  <textarea id="example-text-area"
            class="ui-textarea"
            rows="2"
            placeholder="The <textarea> HTML element represents a multi-line plain-text editing control, useful when you want to allow users to enter a sizeable amount of free-form text, for example a comment on a review or feedback form."
  >
  </textarea>
  <p class="ui-text-p3 text-dark-grey">
    Example of additional text beneath the textarea.
  </p>
</div>
`;

exports[`Styles/Forms Toggles smoke-test 1`] = `
<div class="grid sm:grid-cols-2 gap-4">
  <div class="flex gap-4 items-center">
    <label class="ui-toggle">
      <input type="checkbox">
      <span class="ui-toggle-slider">
      </span>
    </label>
    <p>
      Default
    </p>
  </div>
  <div class="flex gap-4 items-center">
    <label class="ui-toggle">
      <input type="checkbox"
             checked
      >
      <span class="ui-toggle-slider">
      </span>
    </label>
    <p>
      Pre-checked
    </p>
  </div>
  <div class="flex gap-4 items-center">
    <label class="ui-toggle">
      <input type="checkbox"
             disabled
      >
      <span class="ui-toggle-slider">
      </span>
    </label>
    <p>
      Disabled
    </p>
  </div>
  <div class="flex gap-4 items-center">
    <label class="ui-toggle">
      <input type="checkbox"
             class="peer"
      >
      <span class="ui-toggle-slider peer-checked:!bg-pink-600 bg-blue-800">
      </span>
    </label>
    <p>
      Custom color
    </p>
  </div>
</div>
`;
